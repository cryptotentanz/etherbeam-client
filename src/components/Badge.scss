@use '../styles/resources/sizes';

.my-badge {
  border-width: sizes.$border;
  border-style: solid;
  border-radius: sizes.$border-radius-s;
  padding: 0px sizes.$spacer-xs;
}
