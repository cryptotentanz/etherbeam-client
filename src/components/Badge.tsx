import React, { FC, ReactNode } from 'react'
import './Badge.scss'

type ColorProp = 'dark' | 'light' | 'primary' | 'secondary' | 'positive' | 'negative'
type SizeProp = 's' | 'm' | 'l' | 'xl'

interface BadgeProps {
  color?: ColorProp
  size?: SizeProp
  className?: string
  children?: ReactNode
}

const Badge: FC<BadgeProps> = ({ color = 'light', size, className, children }: BadgeProps) => {
  const spanClassName = [`my-badge my-${color}-bg`, size ? `my-size-${size}` : '', className].join(' ')

  return <span className={spanClassName}>{children}</span>
}

export default Badge
