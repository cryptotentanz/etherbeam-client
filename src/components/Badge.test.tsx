import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Badge from './Badge'

describe('<Badge />', () => {
  const content = 'Content.'
  const classNane = 'my-class'

  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Badge>{content}</Badge>)
    })

    it('should render', () => {
      const span = component.find('span').at(0)
      expect(span.hasClass('my-badge')).toBeTruthy()
      expect(span.hasClass('my-light-bg')).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Color', () => {
    beforeAll(() => {
      component = shallow(<Badge color="primary">{content}</Badge>)
    })

    it('should render', () => expect(component.find('span').at(0).hasClass('my-primary-bg')).toBeTruthy())
  })

  describe('Size', () => {
    beforeAll(() => {
      component = shallow(<Badge size="l">{content}</Badge>)
    })

    it('should render', () => expect(component.find('span').at(0).hasClass('my-size-l')).toBeTruthy())
  })

  describe('Class', () => {
    beforeAll(() => {
      component = shallow(<Badge className={classNane}>{content}</Badge>)
    })

    it('should render', () => expect(component.find('span').at(0).hasClass(classNane)).toBeTruthy())
  })
})
