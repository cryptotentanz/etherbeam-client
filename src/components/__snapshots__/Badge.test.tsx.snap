// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> Default should match snapshot 1`] = `
<span
  class="my-badge my-light-bg  "
>
  Content.
</span>
`;
